x/oracle/module/genesis.go

```diff
@@ -9,30 +9,51 @@ import (
 
 // InitGenesis initializes the module's state from a provided genesis state.
 func InitGenesis(ctx sdk.Context, k keeper.Keeper, genState types.GenesisState) {
-	// this line is used by starport scaffolding # genesis/module/init
-	k.SetPort(ctx, genState.PortId)
-	// Only try to bind to port if it is not already bound, since we may already own
-	// port capability from capability InitGenesis
-	if k.ShouldBound(ctx, genState.PortId) {
-		// module binds to the port on InitChain
-		// and claims the returned capability
-		err := k.BindPort(ctx, genState.PortId)
-		if err != nil {
-			panic("could not claim port capability: " + err.Error())
-		}
-	}
 	if err := k.SetParams(ctx, genState.Params); err != nil {
+		// TODO: should we panic here
 		panic(err)
 	}
+
+	for _, bandPriceState := range genState.BandPriceStates {
+		k.SetBandPriceState(ctx, bandPriceState.Symbol, bandPriceState)
+	}
+
+	for _, bandOracleRequest := range genState.BandOracleRequests {
+		k.SetBandOracleRequest(ctx, *bandOracleRequest)
+	}
+
+	k.SetBandParams(ctx, genState.BandParams)
+
+	if genState.BandParams.IbcPortId != "" {
+		k.SetPort(ctx, genState.BandParams.IbcPortId)
+		// Only try to bind to port if it is not already bound, since we may already own port capability
+		if !k.IsBound(ctx, genState.BandParams.IbcPortId) {
+			// module binds to the port on InitChain
+			// and claims the returned capability
+			err := k.BindPort(ctx, genState.BandParams.IbcPortId)
+			if err != nil {
+				panic(types.ErrBandPortBind.Error() + err.Error())
+			}
+		}
+	}
+ 
+	if genState.BandLatestClientId != 0 {
```

Comment:
why we need clientID in state ?

Reply:
ClientId is unique identifier for each request, we have to save that to know the each pair of request and response from BandProtocol, the document for it [here](https://docs.bandchain.org/core-concepts/cosmos-ibc)